localstack/services/awslambda/plugins.py

```diff
@@ -20,3 +26,18 @@ def lambda_java_libs() -> Package:
     from localstack.services.awslambda.packages import lambda_java_libs_package
 
     return lambda_java_libs_package
+
+
+@hooks.on_infra_start()
+def validate_configuration() -> None:
+    if LAMBDA_EXECUTOR == "local":
```

Comment:
any reason this only applies to `local` ?

Reply:
`LAMBDA_EXECUTOR` is part of the regular deprecation warning and this is an emphasis for the local executor because it does not require the Docker socket mounting.
We can easily extend the additional warning for other values as well.